cmd/argocd/commands/app.go

```diff
@@ -525,43 +522,30 @@ func NewApplicationDiffCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 			appName := args[0]
 			app, err := appIf.Get(context.Background(), &application.ApplicationQuery{Name: &appName, Refresh: refresh})
 			errors.CheckError(err)
-			liveObjs, err := app.Status.ComparisonResult.LiveObjects()
+
+			compareObjs, err := app.Status.ComparisonResult.TargetObjects()
 			errors.CheckError(err)
 
-			var compareObjs []*unstructured.Unstructured
-			if local != "" {
-				if env == "" {
-					log.Fatal("--env required when performing local diff")
-				}
-				ksApp, err := ksonnet.NewKsonnetApp(local)
-				errors.CheckError(err)
-				compareObjs, err = ksApp.Show(env)
-				errors.CheckError(err)
-				if len(app.Spec.Source.ComponentParameterOverrides) > 0 {
-					log.Warnf("Unable to display parameter overrides")
-				}
-				compareObjs, liveObjs = diff.MatchObjectLists(compareObjs, liveObjs)
-			} else {
-				if env != "" {
-					log.Fatal("--env option invalid when performing git diff")
-				}
-				compareObjs, err = app.Status.ComparisonResult.TargetObjects()
-				errors.CheckError(err)
-			}
+			liveObjs, err := app.Status.ComparisonResult.LiveObjects()
+			errors.CheckError(err)
 
-			// In order for the diff to be clean, need to set our app labels
-			setAppLabels(appName, compareObjs)
-			diffResults, err := diff.DiffArray(compareObjs, liveObjs)
+			diffManifestResponse, err := appIf.DiffManifests(context.Background(), &application.ApplicationManifestDiffRequest{
```

Comment:
It looks like you removed the local repository comparison feature. Was this intentional?

Reply:
removed